### ISSUE TITLE
Rename MASON_ANDROID_ABI values to match official ABI names

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mason#130]
MASON_ANDROID_ABI currently uses a different naming convention than the [official naming convention](http://developer.android.com/ndk/guides/abis.html) for Android ABIs, which is needlessly confusing. The values should just be the same as the official names.

Concretely, this means the following `MASON_ANDROID_ABI` values should be renamed:
- `arm-v5` ⇢ `armeabi`
- `arm-v7` ⇢ `armeabi-v7a`
- `arm-v8` ⇢ `arm64-v8a`
- `x86-64` ⇢ `x86_64`
- `mips-64` ⇢ `mips64`

`x86` and `mips` are already the same as the official names.

cc @kkaefer @bleege

**Comment by kkaefer** [comment]
Agreed, let's do this. Should we retain the `MASON_ANDROID_PLATFORM`, which is currently `9` or `21` (for 64 bit), or just generally compile against the most recent SDK, and drop it? Retaining it would mean that the `MASON_PLATFORM_VERSION` would be `armeabi-9` or something like that to indicate the Android API level.

**Comment by jfirebaugh** [comment]
I'm fine either way. There was some discussion in chat about whether `9` and `21` were important to mapbox-gl-js in some way, without a clear answer. [The Android documentation](http://developer.android.com/ndk/guides/stable_apis.html) says:

> Each new release of NDK headers and libraries for a given Android API level is cumulative; you are nearly always safe if you use the most recently released headers when building your app. For example, you can use the NDK headers for Android API level 21 for an app targeting API level 16. By doing so, however, you increase your APK's footprint.

**Comment by kkaefer** [comment]
> you increase your APK's footprint.

Wondering what that means; are they referring to binary size? If so, it's unclear to me why that would increase binary size in any significant way...

**Comment by tobrun** [comment]
> Wondering what that means; are they referring to binary size?

Yes this is the binary size, I'm guessing that extra methods will be added to provide backwards compatibility but I'm also not sure if these changes will have significant impact, worth doing a test around.

**Comment by bleege** [comment]
FWIW... we always use the latest Android SDK version on the Java side for compilation.  This keeps things closer to what Google is pushing.  The current version is [`23`](https://github.com/mapbox/mapbox-gl-native/blob/d64b8ad25ca7c2297221337d4021924b5103cc77/platform/android/MapboxGLAndroidSDK/gradle.properties#L16).  On a related note, the Android Minimum that in use is [`15`](https://github.com/mapbox/mapbox-gl-native/blob/d64b8ad25ca7c2297221337d4021924b5103cc77/platform/android/MapboxGLAndroidSDK/gradle.properties#L15).
